builtin/providers/google/resource_compute_backend_service.go

```diff
@@ -0,0 +1,410 @@
+package google
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"regexp"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/hashcode"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"google.golang.org/api/compute/v1"
+	"google.golang.org/api/googleapi"
+)
+
+func resourceComputeBackendService() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceComputeBackendServiceCreate,
+		Read:   resourceComputeBackendServiceRead,
+		Update: resourceComputeBackendServiceUpdate,
+		Delete: resourceComputeBackendServiceDelete,
+
+		Schema: map[string]*schema.Schema{
+			"backend": &schema.Schema{
+				Type: schema.TypeSet,
```

Comment:
In fact `backend` is not a singleton field, hence there's a reason why to keep this as `TypeSet`.

This is typically to prevent issues where API returns the list in different order -> hashes are computed and order doesn't matter.

Otherwise if user had backend `A, B`, inserted as `A, B` (and indexed as `0, 1`), but API suddenly returned `B, A`, terraform would keep trying to change the order because `0` is suddenly B, but was A.